2-mandatory/1-weather-report.js

```diff
@@ -9,55 +9,52 @@
         - return an array of strings, which is a statement about the temperature of each city.
             For example, "The temperature in London is 10 degrees"
         - Hint: you can call the temperatureService function from your function
+        https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
+        npm test -- --testPathPattern mandatory/1-weather-report.js
 */
 
 function getTemperatureReport(cities) {
-    // TODO
+  result = [];
+  for (const i of cities) {
+    result.push(`The temperature in ${i} is ${temperatureService(i)} degrees`);
```

Comment:
The use of the variable name `i` here is misleading. Why do you think this the case? Or, in other words, why might get people get confused by the name `i`?

Reply:
Thanks for comment , i agree with you that variable "i" might be confusing for others . i have changed the name of the  variable to "city"

Reply:
Awesome, nice one!